src/autopas/utils/checkFunctorType.h

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file checkFunctorType.h
+ *
+ * @date 28 Aug 2023
+ * @author muehlhaeusser
+ */
+
+#pragma once
+
+#include "autopas/baseFunctors/PairwiseFunctor.h"
+#include "autopas/baseFunctors/TriwiseFunctor.h"
+
+namespace {
+/**
+ * Returns false for any types that are not a PairwiseFunctor
+ * @return false
+ */
+std::false_type isPairwiseFunctorImpl(...);
+/**
+ * Return true for a PairwiseFunctor type
+ * @tparam ParticleT
+ * @tparam FunctorT
+ * @return true
+ */
+template <typename ParticleT, typename FunctorT>
+std::true_type isPairwiseFunctorImpl(autopas::PairwiseFunctor<ParticleT, FunctorT> const volatile &);
```

Comment:
```suggestion
std::true_type isPairwiseFunctorImpl(const autopas::PairwiseFunctor<ParticleT, FunctorT> &);
```
1. In AutoPas we use `const east` ;)
2. Why `volatile`?